### DISCUSSION TITLE
Why isnt it possible to input .mlp audio files (truehd)

### DISCUSSION THREAD

**Discussion by Mamacito473** [VoidXH/Cavern#138] [Q&A] [answered]
Hey,

sorry for the noobish question but I was wondering why cavern does not handle truehd atmos audio files with 16-ch presentation

thx if anyone replies

**Comment by VoidXH** [comment]
Hello, only DD+ Atmos can be supported, not TrueHD. It would require Dolby to share the documentation for it, which they will likely never do. DD+ is open, because it's a broadcast standard, the same goes for AC-4, but not TrueHD.

**Comment by ValeZAA** [comment]
TrueHD atmos patch with decoder of last substream for ffmpeg https://patchwork.ffmpeg.org/project/ffmpeg/patch/20250322174949.531092-1-eynard.massimo@gmail.com/

**Reply by VoidXH** [reply]
FFmpeg source code can't be used because of a licence mismatch. The document which resulted in this commit is still required for a proper TrueHD implementation. Additionally, object positions are not decoded here, which would make rendering incomplete.

**Reply by ValeZAA** [reply]
There is no document, it is reverse engineered.

extract_objects is the name of the flag

**Reply by ValeZAA** [reply]
Looks like the author understands what is being done, pleade help if you can 🙂

Actually, in current use of MLP/TrueHD when decoding using the fourth substream, reconstructed channels cannot be mapped to single loudspeakers or fixed position and acts more like "tracks" which are feeds for audio objects.
These audio objects are indeed not fixed in space, their position is defined by additional time-varying metadata.

**Comment by VoidXH** [comment, marked as answer]
My old reply aged like milk. Cavernize now fully supports TrueHD through truehdd.

**Reply by Mamacito473** [reply]
great work, much appreciated, wonderful !!! atmos is finally not a pain in the a** to deal with !
